core/src/rigidbody.cpp

```diff
@@ -0,0 +1,289 @@
+void RigidBody::on_destroy() {
+    if (m_dynamic) {
+        m_scene->removeActor(*m_dynamic);
+    } else if (m_static) {
+        m_scene->removeActor(*m_static);
+    }
+}
+
+vec3 RigidBody::get_position() const {
+    vec3 v;
+    if (m_dynamic) {
+        v = vec3(m_dynamic->getGlobalPose().p.x, m_dynamic->getGlobalPose().p.y, m_dynamic->getGlobalPose().p.z);
+
+    } else {
+        v = vec3(m_static->getGlobalPose().p.x, m_static->getGlobalPose().p.y, m_static->getGlobalPose().p.z);
+    }
+    return v;
+}
+
+quat RigidBody::get_rotation() const {
+    quat q;
+    if (m_dynamic) {
+        q = quat(m_dynamic->getGlobalPose().q.w, m_dynamic->getGlobalPose().q.x, m_dynamic->getGlobalPose().q.y,
+                 m_dynamic->getGlobalPose().q.z);
+    } else {
+        q = quat(m_static->getGlobalPose().q.w, m_static->getGlobalPose().q.x, m_static->getGlobalPose().q.y,
+                 m_static->getGlobalPose().q.z);
+    }
+    return q;
+}
+
+physx::PxMaterial* RigidBody::get_pxmaterial() {
+    return m_material;
+}
+
+void RigidBody::set_transform(const vec3& position, const quat& rotation) {
+    if (m_dynamic == NULL && m_static == NULL) {
+        return;
+    }
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
+    physx::PxVec3 p = physx::PxVec3(p_x, p_y, p_z);
+    physx::PxQuat q = physx::PxQuat(r_w, r_x, r_y, r_z);
+    if (m_dynamic) {
+        m_dynamic->setGlobalPose(PxTransform(p));
+        m_dynamic->setGlobalPose(PxTransform(q));
+    } else {
+        m_static->setGlobalPose(PxTransform(p));
+        m_static->setGlobalPose(PxTransform(q));
+    }
+}
+
+void RigidBody::set_position(const vec3& position) {
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    physx::PxVec3 p = physx::PxVec3(p_x, p_y, p_z);
+    if (m_dynamic) {
+        m_dynamic->setGlobalPose(PxTransform(p));
+    } else {
+        m_static->setGlobalPose(PxTransform(p));
+    }
+}
+void RigidBody::set_rotation(const quat& rotation) {
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
+    physx::PxQuat q = physx::PxQuat(r_w, r_x, r_y, r_z);
+    if (m_dynamic) {
+        m_dynamic->setGlobalPose(PxTransform(q));
+    } else {
+        m_static->setGlobalPose(PxTransform(q));
+    }
+}
+
+void RigidBody::set_material(physx::PxMaterial* material) {
+    m_material = material;
+}
+
+void RigidBody::set_physics_and_scene(physx::PxScene* scene, physx::PxPhysics* physics) {
+    m_scene = scene;
+    m_physics = physics;
+}
+
+void RigidBody::create_cube_rigid_dynamic(const vec3& halfSize,
+                                          const float& mass,
+                                          const vec3& position,
+                                          const quat& rotation) {
```

Comment:
Since we are guaranteeing (please correct me if not the case @icewinow @BeardyKing) that any position and rotation that we will need is tied to the `Transform` component, can we delete this arguments and extract the needed values from it?

Something like:
```cpp
    vec3 position(std::numeric_limits<float>::max()); // We don't want it to start at the origin, if it doesn't have a position
    auto sceneOpt = Scene::get_active_scene();
    if (sceneOpt) {
        auto goOpt = sceneOpt->get().get_game_object_from_component(*this);
        if (goOpt) {
            position = goOpt->get_component<components::Transform>().get_position();
        }
    }
```

Or abstract it to its own function:
```cpp
vec3 RigidBody::get_position_from_transform() {
    auto sceneOpt = Scene::get_active_scene();
    if (!sceneOpt) {
        auto goOpt = sceneOpt->get().get_game_object_from_component(*this);
        if (goOpt) {
            return goOpt->get_component<components::Transform>().get_position();
        }
    }
    return vec3(std::numeric_limits<float>::max());
}
```

We can do this with `Transform`, `Name`, `Hierarchy`, and components like those, because we can guarantee that they are always going to exist inside a given GameObject — they are added at creation, and they will not be removable inside the Editor.